src/main/java/net/kaczmarzyk/spring/data/jpa/utils/JsonUtils.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Andrei Shakarov
+ */
+public class JsonUtils {
+    private static final String DOT_DELIMITER = "\\.";
+
+    private JsonUtils() {
+    }
+
+    public static Collection<String> getValuesFromJson(String json, String key) {
```

Comment:
The body will be parsed as many times as many specs are defined - it's inefficient and we should consider some kind of caching as long term solution. However, in my opinion at this moment it's acceptable in the first version of json support.   (I'll create a separate issue for this when this PR will be merged)